### PR TITLE
docs: add test guidelines

### LINKED ISSUE #595: Create tests guidelines.
Create a `TEST.md` that explains how we're testing Gitpoint.
I'll use the test for `NotificationListItem` in this PR as a reference. https://github.com/gitpoint/git-point/pull/589

### PR DESCRIPTION
**Pull request by alejandronanez**
Add TEST guidelines. Let me know what do you think.

Closes #595

### CODE REVIEW

File: TESTS.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Guidelines about how to write your unit tests for Gitpoint
```
Comment (andrewda) [inline TESTS.md]: Keep the title short. Maybe just `Test Guidelines`?

File: TESTS.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        isOpen={false}
+      />
+    );
+
+    expect(wrapper.find(Bar).length).toBe(1);
+  });
+
+  it('should not render the Bar component when isOpen is falsy', () => { ... });
+
+  it('should return 'the NAME' when foo received \'name\' as a param', () => { ... });
```
Comment (andrewda) [inline TESTS.md]: Either `... return \'the NAME\' ...` or use double quotes and get rid of the backslashes for `... \'name\' ...`

File: TESTS.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Test guidlines
```
Comment (andrewda) [inline TESTS.md]: Tiny typo: should be "Guidelines". Let's also use title case here (capital "G").

**Comment by housseindjirdeh** [comment]
Oh man thank you @alejandronanez, this was really needed <3

**Pull request merged**